### PR TITLE
#14 - Add proper support for string starts/ends operators

### PR DESCRIPTION
**Pull request by roele**
**1. Issue Link:** 
https://github.com/intuit/graphql-filter-java/issues/14

**2. Brief explanation of a change:** 
Implement START/END string operators instead of falling through CONTAINS.

**3. Will it break existing clients and code in production?**
Can possibly be considered a breaking change as behaviour will be more restrictive with this change.

### CODE REVIEW

File: src/main/java/com/intuit/graphql/filter/visitors/SQLExpressionVisitor.java [hunk 1]
```diff
@@ -162,7 +162,11 @@ public String visitExpressionValue(ExpressionValue<? extends Comparable> value,
         Operator operator = operatorStack.pop();
         List<? extends Comparable> expressionValues  = value.getValues();
 
-        if (operator == Operator.CONTAINS || operator == Operator.STARTS || operator == Operator.ENDS ) {
+        if (operator == Operator.STARTS) {
+            expressionBuilder.append("'").append(expressionValues.get(0)).append("%").append("'");
+        } else if (operator == Operator.ENDS) {
```
Comment (jaiswalsk) [inline src/main/java/com/intuit/graphql/filter/visitors/SQLExpressionVisitor.java:167]: Lets add tests for these operators in the SQLExpressionTest file.

File: src/main/java/com/intuit/graphql/filter/visitors/JpaSpecificationExpressionVisitor.java [hunk 2]
```diff
@@ -118,7 +118,13 @@ public Predicate toPredicate(Root<T> root, CriteriaQuery<?> criteriaQuery, Crite
                 switch (binaryExpression.getOperator()) {
                     /* String operations.*/
                     case STARTS:
+                        predicate = criteriaBuilder.like(path, operandValue.value() + "%");
+                        break;
+
                     case ENDS:
+                        predicate = criteriaBuilder.like(path, "%" + operandValue.value());
+                        break;
```
Comment (jaiswalsk) [inline src/main/java/com/intuit/graphql/filter/visitors/JpaSpecificationExpressionVisitor.java:126]: Thanks for adding these cases.

**Review by jaiswalsk — APPROVED** [review]

File: src/main/java/com/intuit/graphql/filter/visitors/SQLExpressionVisitor.java (hunk 1 above)
Comment (roele) [reply src/main/java/com/intuit/graphql/filter/visitors/SQLExpressionVisitor.java:167]: 👍  Added a test for each of those.

**Review by jaiswalsk — APPROVED** [review]

**Comment by jaiswalsk** [comment]
Looks Good! 🚀  Thanks @roele for your contributions.

**Pull request merged**